lib/bitvec.cpp

```diff
@@ -78,6 +80,8 @@ bitvec &bitvec::operator<<=(size_t count) {
 }
 
 bitvec bitvec::getslice(size_t idx, size_t sz) const {
+    assert(static_cast<int>(idx) >= 0);
+    assert(static_cast<int>(sz) >= 0);
```

Comment:
calling getslice with `sz == (size_t)-1` is a perfectly valid -- it means get from idx to the end of the bitvec.  Equivalent to making a copy then right-shifting by index.